update.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+msg () {
+	1>&2 echo $*
+}
+
+# check that we are in a prototype
+if ! grep -q govuk-prototype-kit package.json 2> /dev/null; then
+	msg 'ERROR you must run update.sh in a folder containing a GOV.UK Prototype Kit installation'
+	msg 'Exiting'
+	exit 1
+fi
+
+# make the update folder
+mkdir -p update
+# stop git from committing the update folder
+echo '*' > update/.gitignore
+
+{
+  # download the latest Prototype Kit zip file to the update folder
+  cd update
+
+  if ! ls govuk-prototype-kit*.zip > /dev/null 2>&1; then
```

Comment:
I'm not sure what value the check here and on line 38 give? Without them the code is simpler, and does exactly the same thing each time you run it. I think there's a danger that with the checks, if anything goes wrong with the zip or update folder, these checks mean the process could fail

Reply:
It provides hooks into the process, either for testing or for customization by an expert user. For instance, I could put a test zip file `govuk-prototype-kit-99.0.0.zip` in the update folder and then the script would use that.

Reply:
thats a good point, but for example if the script fails part way through for any reason, and leaves a corrupt zip or update folder, it cannot recover as is

Reply:
Hmm, good point!